src/main/java/com/microsoft/azure/spring/data/documentdb/repository/support/SimpleDocumentDbRepository.java

```diff
@@ -43,16 +43,22 @@ public SimpleDocumentDbRepository(DocumentDbEntityInformation<T, ID> metadata,
     @Override
     public <S extends T> S save(S entity) {
         Assert.notNull(entity, "entity must not be null");
-
         // create collection if not exists
         documentDbOperations.createCollectionIfNotExists(entityInformation.getCollectionName(),
                 entityInformation.getPartitionKeyFieldName(),
                 entityInformation.getRequestUint());
 
         // save entity
-        documentDbOperations.insert(entityInformation.getCollectionName(),
-                entity,
-                entityInformation.getPartitionKeyFieldValue(entity));
+        if (entityInformation.isNew(entity)) {
```

Comment:
no need to override IsNewStrategy()?

Reply:
Currently only use the default implementation of Spring Data Commons, only check whether the ID is empty or equals 0.